File: src/main/java/org/eln/eln3/sim/process/heater/ResistorHeatThermalLoad.java
package org.eln.eln3.sim.process.heater;

import org.eln.eln3.sim.IProcess;
import org.eln.eln3.sim.ThermalLoad;
import org.eln.eln3.sim.mna.component.Resistor;

public class ResistorHeatThermalLoad implements IProcess {

    Resistor resistor;
    ThermalLoad load;

    public ResistorHeatThermalLoad(Resistor r, ThermalLoad load) {
        this.resistor = r;
        this.load = load;
    }

    @Override
    public void process(double time) {
        load.movePowerTo(resistor.getPower());
    }
}

File: src/main/kotlin/org/eln/eln3/sim/process/thermal/ResistorHeatThermalLoad.kt
package org.eln.eln3.sim.process.thermal

import org.eln.eln3.sim.IProcess
import org.eln.eln3.sim.thermal.ThermalLoad
import org.eln.eln3.sim.mna.component.Resistor

class ResistorHeatThermalLoad(var resistor: Resistor, var load: ThermalLoad) : IProcess {
    override fun process(time: Double) {
        load.addThermalPower(resistor.power)
    }
}